tests/test_data/test_shape.py

```diff
@@ -17,14 +17,243 @@
+    shape = Shape.polygonFromGeoJson(data_true_geoJson)
+    assert isinstance(shape, typ)
+
+    # input typechecks
+    data_true_geoJson = {"type": "Feature",
+                         "geometry": data,
+                         "properties": {"name": "Dinagat Islands"}}
+    caplog.clear()
+    shape = Shape.polygonFromGeoJson([(0, 1), (1, 0), (0, 1)])
+    assert shape is None
+    assert "Input has to be dict" in caplog.text
+
+    # error if not (Multipolygon)
+    caplog.clear()
+    shape = Shape.polygonFromGeoJson({"type": "Point", "coordinates": [125.6, 10.1]})
+    assert shape is None
+    assert "Not a valid GeoJson" in caplog.text
+
+    # Err if path does not exist
+    caplog.clear()
+    shape = Shape.polygonFromGeoJson("nopath/to/file")
+    assert "does not exist" in caplog.text
+
+
+@pytest.mark.parametrize("fn",
+    ["simple_Fylker_0.json", "simple_Kommuner_0.json", "simple_Kommuner_291.json"]) # noqa
+@pytest.mark.data
+def testDataShape_GeoJsonFromPolygon(fn, tmpConf, tmpDir, caplog):
+    """ Checks that geojson file is stored and Shape instance is created"""
+    dataPath = Path(tmpDir) / "data"
+    dataPath.mkdir(exist_ok=True)
+    tmpConf.dataPath = dataPath
+
+    fn = Path("tests/test_data/data") / fn
+    with fn.open() as f:
+        data = json.load(f)
+        data = data["polygon"]
+
+    # plain comparison
+    data = geometry.shape(data)
+    expected = {"type": "Feature", "geometry": geometry.mapping(data)}
+    result = Shape.geoJsonFromPolygon(data)
+    assert result == expected
+
+    # added arguments
+    pars = {"parameters": {"name": "foo"}, "crs" : 123}
+    expected = {"type": "Feature", "geometry": geometry.mapping(data),
+                "parameters": {"name": "foo"}, "crs" : 123}
+    result = Shape.geoJsonFromPolygon(data, pars)
+    assert result == expected
+
+    # warn if special key
+    pars = {"geometry": 123, "crs" : 123}
+    expected = {"type": "Feature", "geometry": geometry.mapping(data)}
+    caplog.clear()
+    result = Shape.geoJsonFromPolygon(data, pars)
+    assert result == expected
+    assert "Cannot append" in caplog.text
+
+
+@pytest.mark.data
+@pytest.mark.parametrize("uuid,expected,logtext",
+    [(None, False, "is not a string"), # noqa # not a string
+     ('33d0c48f-b58c-4b1a-b224-e93b03393cb3', True, ""), # gen by uuid4
+     ('33d0c48fb58c4b1ab224e93b03393cb3', False, ""), # missing '-' character
+     ('0', False, "ValueError")]) # check catching error
+def testDataShape_ValidateUudi4(uuid, expected, logtext, caplog, tmpConf, tmpDir):
+    """Test that the UUIDs are validated or errors are logged.
+    """
+    dataPath = Path(tmpDir) / "data"
+    dataPath.mkdir(exist_ok=True)
+    tmpConf.dataPath = dataPath
+
+    shape = Shape(uuid)
+    caplog.clear()
 
+    result = shape._validateUuid()
+    assert result == expected
+    if not result:
+        assert logtext in caplog.text
 # END Test testDataShape_Init
```

Comment:
If you don't add these comments below tests, at least remove the one that doesn't match.